File: core-js/src/elements/Link.ts
import { Matrix4x4 } from '../math/Matrix4x4';

export class Link {
  public id: string;
  /** Fixed static offset this link represents in the kinematic chain */
  public transform: Matrix4x4;

  constructor(id: string, transform?: Matrix4x4) {
    this.id = id;
    this.transform = transform ?? new Matrix4x4();
  }
}

File: core-py/src/wigglesolve/elements/link.py
from __future__ import annotations
from typing import Optional
from wigglesolve.math.matrix4x4 import Matrix4x4


class Link:
    def __init__(self, id: str, transform: Optional[Matrix4x4] = None) -> None:
        self.id = id
        self.transform = transform if transform is not None else Matrix4x4()
